src/styles/atoms/_spacing.scss

```diff
@@ -0,0 +1,52 @@
+@import '../utils/scale';
+
+$space-scales: -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10;
+$space-scale-factor: 1.618; // golden rule
+
+
+
+// $_breakpoint-mapping: (
+//   xs-only: $breakpoint-xs-only,
+//   xs: all,
+//   sm-only: $breakpoint-sm-only,
+//   sm: $breakpoint-sm-up,
+//   md-only: $breakpoint-md-only,
+//   md: $breakpoint-md-up,
+//   lg-only: $breakpoint-lg-only,
+//   lg: $breakpoint-lg-up,
+//   xl-only: $breakpoint-xl-up
+// );
+
+@mixin _inset($type, $dirs, $step) {
+
+  $dirs: if($dirs == all, top right bottom left, $dirs);
+
+  // validation
+  @each $dir in $dirs {
+    @if index(top right bottom left, $dir) == null {
+      @warn 'supplied first argument contains invalid element `#{$dir}`';
+    }
+  }
+
+  @if index(join($space-scales, reset), $step) == null {
+    @warn 'supplied second argument `#{$step}` is not supported for `step`';
+  }
+
+
+  // calculate rem value
+  $value: if($step == reset, 0, scale($step, $space-scale-factor));
+
+    @each $dir in $dirs {
+      #{$type}-#{$dir}: $value;
+    }
+
+
```

Comment:
unnecessary empty lines

Reply:
@henrylatham one more thing. It is pure horror to review huge PR-s. Please in future separate to smaller ones 🙏 

<img width="811" alt="Screen Shot 2019-11-14 at 15 48 48" src="https://user-images.githubusercontent.com/18653950/68869299-40ae2780-06f9-11ea-8c39-0d3a7ce7df45.png">

In this case almost  each your commit could be separated PR